### DISCUSSION TITLE
How do I mark multiple files so that I can delete, cut, and move them all at once?

### DISCUSSION THREAD

**Discussion by lee-shun** [nvim-tree/nvim-tree.lua#2033] [Q&A] [answered]
How do I mark multiple files so that I can delete, cut, and move them all at once?

**Comment by AlphabetsAlphabets** [comment]
+1

I would also like to know how to do this.

**Comment by lee-shun** [comment]
I take the reference of #1804: 
I modify the code and add keymaps.
you can add the following code to your `on_attach`function:
```lua
			local bulk_trash = function()
				local nodes = api.marks.list()
				if next(nodes) == nil then
					table.insert(nodes, api.get_node_under_cursor())
				end
				vim.ui.input({ prompt = string.format("Trash %s files? [y/n] ", #nodes) }, function(input)
					if input == "y" then
						for _, node in ipairs(nodes) do
							vim.fn.jobstart("trash-put " .. node.absolute_path)
						end
						api.tree.reload()
					end
				end)
			end

			vim.keymap.set("n", "bmr", bulk_trash, opts("Bulk Trash Files"))

```

it works for me:)

**Comment by alex-courtis** [comment]
That's really nice! Could you please add that to the [wiki: Recipes](https://github.com/nvim-tree/nvim-tree.lua/wiki/Recipes)

**Comment by alex-courtis** [comment]
You could call `api.fs.trash` for each node, rather than starting a job / reloading

**Reply by lee-shun** [reply, marked as answer]
Hey! guys,

 I am glad to show you how I write my `on_attach` file to realize this **marked file operations**!

please see the wiki:
https://github.com/nvim-tree/nvim-tree.lua/wiki/Recipes#modify-you-on_attach-function-to-have-ability-to-operate-multiple-files-at-once
